plugins/BEdita/API/tests/IntegrationTest/DefaultValuesTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2019 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\API\Test\IntegrationTest;
+
+use BEdita\API\TestSuite\IntegrationTestCase;
+use Cake\Core\Configure;
+
+/**
+ * Test `DefaultValues` config upon object creation
+ */
+class DefaultValuesTest extends IntegrationTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    public $fixtures = [
+        'plugin.BEdita/Core.streams',
+    ];
+
+    /**
+     * Data provider for testCreate
+     *
+     * @return array
+     */
+    public function createProvider()
+    {
+        return [
+            'files on' => [
+                ['status' => 'on'],
+                'files',
+                ['title' => 'My File'],
+                [
+                    'files' => [
+                        'status' => 'on',
+                    ],
+                ],
+            ],
+
+            'files draft' => [
+                ['status' => 'draft'],
+                'files',
+                ['title' => 'My File'],
+                [],
+            ],
+
+            'folders description' => [
+                ['description' => 'some text'],
+                'folders',
+                ['title' => 'My Folder'],
+                [
+                    'folders' => [
+                        'description' => 'some text',
+                    ],
+                ],
+            ],
+
+            'locations country' => [
+                ['country_name' => 'Italy'],
+                'locations',
+                ['title' => 'My Location'],
+                [
+                    'locations' => [
+                        'country_name' => 'Italy',
+                    ],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Test create a new object and verify DefaultValues.
+     *
+     * @param array $expected Expected result.
+     * @param string $type Object type.
+     * @param array $attributes Object attributes.
+     * @param array $config DefaultValues config.
+     * @return void
+     *
+     * @dataProvider createProvider
+     * @coversNothing
+     */
+    public function testCreate(array $expected, string $type, array $attributes, array $config)
```

Comment:
[minor] Since we’re starting using PHP 7 parameter type hinting (e.g. `string $type`), we could also add return types hinting:

```php
public function createProvider(): array
{
    // …
}

public function testCreate(/* … */): void
{
    // …
}
```

Reply:
Done